BaseFiles/Common/html/pages/control/widgets/homegenie/generic/doorlock.js

```diff
@@ -0,0 +1,85 @@
+﻿[{
+    Name: "Security Alarm System",
+    Author: "Generoso Martello",
```

Comment:
true, there are many places where I should have done something different. I had the code in a rough version since revision 372 - close to a half a year - but didn't get a chance to finish it and also it took me a few days to adapt the code to the new changes - I didn't want to wait more and then have to catch-up again :)